src/Sulu/Bundle/TrashBundle/Application/TrashManager/TrashManager.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\TrashBundle\Application\TrashManager;
+
+use Sulu\Bundle\ActivityBundle\Application\Collector\DomainEventCollectorInterface;
+use Sulu\Bundle\TrashBundle\Application\TrashItemHandler\RestoreTrashItemHandlerInterface;
+use Sulu\Bundle\TrashBundle\Application\TrashItemHandler\StoreTrashItemHandlerInterface;
+use Sulu\Bundle\TrashBundle\Domain\Event\TrashItemCreatedEvent;
+use Sulu\Bundle\TrashBundle\Domain\Event\TrashItemRemovedEvent;
+use Sulu\Bundle\TrashBundle\Domain\Event\TrashItemRestoredEvent;
+use Sulu\Bundle\TrashBundle\Domain\Exception\RestoreTrashItemHandlerNotFoundException;
+use Sulu\Bundle\TrashBundle\Domain\Exception\StoreTrashItemHandlerNotFoundException;
+use Sulu\Bundle\TrashBundle\Domain\Model\TrashItemInterface;
+use Sulu\Bundle\TrashBundle\Domain\Repository\TrashItemRepositoryInterface;
+
+class TrashManager implements TrashManagerInterface
```

Comment:
Would mark this service as final as extending is no official extension point here:

```suggestion
final class TrashManager implements TrashManagerInterface
```